py/stream.c

```diff
@@ -545,16 +555,8 @@ ssize_t mp_stream_posix_read(void *stream, void *buf, size_t len) {
 }
 
 off_t mp_stream_posix_lseek(void *stream, off_t offset, int whence) {
-    const mp_obj_base_t *o = stream;
-    const mp_stream_p_t *stream_p = MP_OBJ_TYPE_GET_SLOT(o->type, protocol);
-    struct mp_stream_seek_t seek_s;
-    seek_s.offset = offset;
-    seek_s.whence = whence;
-    mp_uint_t res = stream_p->ioctl(MP_OBJ_FROM_PTR(stream), MP_STREAM_SEEK, (mp_uint_t)(uintptr_t)&seek_s, &errno);
-    if (res == MP_STREAM_ERROR) {
-        return -1;
-    }
-    return seek_s.offset;
+    int error;
+    return mp_stream_seek(MP_OBJ_FROM_PTR(stream), offset, whence, &error);
```

Comment:
It may be that you need to convert the type `mp_off_t` to `off_t` to get this working.  Something like this:
```c
mp_off_t res = mp_stream_seek(...);
if (res == (mp_off_t)-1) {
    return -1;
}
return res;
```

Reply:
No it didn't fix it. It seems to be expecting a different errorno ?

```
OSError(2,)
```
Exp:
```
OSError(1001,)
```